packages/mui-base/src/InputUnstyled/InputUnstyled.tsx

```diff
@@ -119,27 +126,32 @@ const InputUnstyled = React.forwardRef(function InputUnstyled(
   };
 
   const Root = component ?? components.Root ?? 'div';
-  const rootProps = appendOwnerState(
+  const rootComponentsProps = resolveComponentProps(componentsProps.root, ownerState);
+  const rootProps: WithOptionalOwnerState<InputUnstyledRootSlotProps> = appendOwnerState(
     Root,
     {
       ...getRootProps({ ...other, ...componentsProps.root }),
```

Comment:
```suggestion
      ...getRootProps({ ...other, ...rootComponentsProps }),
```

Shouldn't we use here the resolved component's props?